trilinos +teko +tpetra build fails: cmake: teko requires tpetra enable GlobalOrdinal

`trilinos +teko +tpetra +stratimikos` build fails using:
 * `spack@develop` (7b6ca5903861f7b7e648d651f5715ba6262e2a58 from `Sat Jun 19 08:25:21 2021 -0500`):

Concrete spec: [trilinos.spec.yaml.txt](https://github.com/spack/spack/files/6689160/trilinos.spec.yaml.txt)
Build log: [trilinos-build-out.txt](https://github.com/spack/spack/files/6689159/trilinos-build-out.txt)

```
$> spack install -f ./trilinos.spec.yaml
...
==> Installing trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj
==> Using cached archive: /spack/var/spack/cache/_source-cache/git//trilinos/Trilinos.git/4796b92fb0644ba8c531dd9953e7a4878b05c62d.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> No patches needed for trilinos
==> trilinos: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/boost-1.76.0-xfahzpfzjei2i62ni7seuszkjwfzmvjz/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/bzip2-1.0.8-y67eszd5llhcqotum46vjsifxuxao3y7/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-nawpdrhj3qspz2o4tb7pc6mk2cafuklj/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.4.1-q6hzuh4lmnlthnlqphvvyuk557hej35c/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-ux5xkhgfhhmitpkxfuf5h6ooudsiwhd4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxml2-2.9.10-qlt5ajqqmtin3atqcycnv67oex63cvny/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libiconv-1.16-w6zptbclncs3jsnl3i4qtg72fgheiwdf/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xz-5.2.5-ynkyqliadxgrqpk4mavevv3onehjrdco/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/zlib-1.2.11-p7dmb5pqi6svirhovljrkp3uyfiytz3m/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/ncurses-6.2-n5vhymfigwg6e45k7thdfyeswuhxojtx/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libfabric-1.12.1-lgug4cxazbl24jdtmttovd5qpnr6os7z/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hdf5-1.12.0-zlpe7nx7g3rt4sthx36aoqzysrwyrjg4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-vw2cdve2xpvtcq2r32ogoptxd4cyjfre/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hypre-2.20.0-btbzpjcwalrwtdk62idk2qzzjioeylxs/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/openblas-0.3.15-3uzmsi2jrh4o7rcrys4wa2tg6ovki55t/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/matio-1.5.17-kzserul5bk24dwg2cr4zep4dvmofmgiz/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/metis-5.1.0-lxkd2thzi5xc5z3aqvw4eodot57r53y5/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mumps-5.4.0-vy5bpjrdcgs63m47fjy3fbfxgo6frnel/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/netlib-scalapack-2.1.0-ryeq4ydyoueo2triza7dhwg6ihkkks46/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/netcdf-c-4.8.0-rudvlpfj65jss3v6byafnuqzctjx2e7f/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/parmetis-4.0.3-rdrwwqrizoiz4h3xiehqqwfzimpuv5zh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/suite-sparse-5.10.1-375laj47cael7c3xlphppxz6ljxos3vp/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/gmp-6.2.1-jjastjjnwyepukurcek34ileelgrpufo/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpfr-4.1.0-3mdlqs3qnt5aowz6pdozaw5bmpvw7p2s/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/glm-0.9.9.8-53yq3o2mplgjk6yiy2hmcud2iigfwkqt/lib64' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/boost-1.76.0-xfahzpfzjei2i62ni7seuszkjwfzmvjz;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/bzip2-1.0.8-y67eszd5llhcqotum46vjsifxuxao3y7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.20.3-57eqw4fz35eug5zhivffh2ry2mfagbu3;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/glm-0.9.9.8-53yq3o2mplgjk6yiy2hmcud2iigfwkqt;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/gmp-6.2.1-jjastjjnwyepukurcek34ileelgrpufo;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hdf5-1.12.0-zlpe7nx7g3rt4sthx36aoqzysrwyrjg4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.4.1-q6hzuh4lmnlthnlqphvvyuk557hej35c;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hypre-2.20.0-btbzpjcwalrwtdk62idk2qzzjioeylxs;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libfabric-1.12.1-lgug4cxazbl24jdtmttovd5qpnr6os7z;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libiconv-1.16-w6zptbclncs3jsnl3i4qtg72fgheiwdf;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-ux5xkhgfhhmitpkxfuf5h6ooudsiwhd4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxml2-2.9.10-qlt5ajqqmtin3atqcycnv67oex63cvny;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/matio-1.5.17-kzserul5bk24dwg2cr4zep4dvmofmgiz;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/metis-5.1.0-lxkd2thzi5xc5z3aqvw4eodot57r53y5;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpfr-4.1.0-3mdlqs3qnt5aowz6pdozaw5bmpvw7p2s;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-nawpdrhj3qspz2o4tb7pc6mk2cafuklj;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mumps-5.4.0-vy5bpjrdcgs63m47fjy3fbfxgo6frnel;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/ncurses-6.2-n5vhymfigwg6e45k7thdfyeswuhxojtx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/netcdf-c-4.8.0-rudvlpfj65jss3v6byafnuqzctjx2e7f;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/netlib-scalapack-2.1.0-ryeq4ydyoueo2triza7dhwg6ihkkks46;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-vw2cdve2xpvtcq2r32ogoptxd4cyjfre;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/openblas-0.3.15-3uzmsi2jrh4o7rcrys4wa2tg6ovki55t;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/parmetis-4.0.3-rdrwwqrizoiz4h3xiehqqwfzimpuv5zh;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/suite-sparse-5.10.1-375laj47cael7c3xlphppxz6ljxos3vp;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xz-5.2.5-ynkyqliadxgrqpk4mavevv3onehjrdco;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/zlib-1.2.11-p7dmb5pqi6svirhovljrkp3uyfiytz3m' '-DTrilinos_VERBOSE_CONFIGURE:BOOL=OFF' '-DTrilinos_ENABLE_TESTS:BOOL=OFF' '-DTrilinos_ENABLE_EXAMPLES:BOOL=OFF' '-DTrilinos_ENABLE_CXX11:BOOL=ON' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DTrilinos_ENABLE_DEBUG:BOOL=OFF' '-DTPL_ENABLE_MPI:BOOL=ON' '-DCMAKE_C_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-nawpdrhj3qspz2o4tb7pc6mk2cafuklj/bin/mpicc' '-DCMAKE_CXX_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-nawpdrhj3qspz2o4tb7pc6mk2cafuklj/bin/mpic++' '-DCMAKE_Fortran_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-nawpdrhj3qspz2o4tb7pc6mk2cafuklj/bin/mpif90' '-DMPI_BASE_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-nawpdrhj3qspz2o4tb7pc6mk2cafuklj' '-DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF' '-DTrilinos_ENABLE_Amesos:BOOL=ON' '-DTrilinos_ENABLE_Amesos2:BOOL=ON' '-DTrilinos_ENABLE_Anasazi:BOOL=ON' '-DTrilinos_ENABLE_AztecOO:BOOL=ON' '-DTrilinos_ENABLE_Belos:BOOL=ON' '-DTrilinos_ENABLE_Epetra:BOOL=ON' '-DTrilinos_ENABLE_EpetraExt:BOOL=ON' '-DTrilinos_ENABLE_Ifpack:BOOL=ON' '-DTrilinos_ENABLE_Ifpack2:BOOL=ON' '-DTrilinos_ENABLE_Intrepid:BOOL=OFF' '-DTrilinos_ENABLE_Intrepid2:BOOL=OFF' '-DTrilinos_ENABLE_Isorropia:BOOL=OFF' '-DTrilinos_ENABLE_Kokkos:BOOL=ON' '-DTrilinos_ENABLE_MiniTensor:BOOL=OFF' '-DTrilinos_ENABLE_Mesquite:BOOL=OFF' '-DTrilinos_ENABLE_ML:BOOL=ON' '-DTrilinos_ENABLE_MueLu:BOOL=ON' '-DTrilinos_ENABLE_NOX:BOOL=OFF' '-DTrilinos_ENABLE_Piro:BOOL=OFF' '-DTrilinos_ENABLE_Phalanx:BOOL=OFF' '-DTrilinos_ENABLE_PyTrilinos:BOOL=OFF' '-DTrilinos_ENABLE_ROL:BOOL=OFF' '-DTrilinos_ENABLE_Rythmos:BOOL=OFF' '-DTrilinos_ENABLE_Sacado:BOOL=ON' '-DTrilinos_ENABLE_SCOREC:BOOL=OFF' '-DTrilinos_ENABLE_Shards:BOOL=OFF' '-DTrilinos_ENABLE_ShyLU:BOOL=OFF' '-DTrilinos_ENABLE_STK:BOOL=OFF' '-DTrilinos_ENABLE_Stokhos:BOOL=OFF' '-DTrilinos_ENABLE_Stratimikos:BOOL=ON' '-DTrilinos_ENABLE_Teko:BOOL=ON' '-DTrilinos_ENABLE_Tempus:BOOL=OFF' '-DTrilinos_ENABLE_Teuchos:BOOL=ON' '-DTrilinos_ENABLE_Tpetra:BOOL=ON' '-DTrilinos_ENABLE_TrilinosCouplings:BOOL=OFF' '-DTrilinos_ENABLE_Zoltan:BOOL=ON' '-DTrilinos_ENABLE_Zoltan2:BOOL=ON' '-DEpetraExt_BUILD_BTF:BOOL=OFF' '-DEpetraExt_BUILD_EXPERIMENTAL:BOOL=OFF' '-DEpetraExt_BUILD_GRAPH_REORDERINGS:BOOL=OFF' '-DAmesos2_ENABLE_Basker:BOOL=OFF' '-DUSE_XSDK_DEFAULTS:BOOL=OFF' '-DTrilinos_ENABLE_SEACAS:BOOL=ON' '-DTrilinos_ENABLE_SEACASExodus:BOOL=ON' '-DTrilinos_ENABLE_SEACASIoss:BOOL=ON' '-DTrilinos_ENABLE_SEACASEpu:BOOL=ON' '-DTrilinos_ENABLE_SEACASExodiff:BOOL=ON' '-DTrilinos_ENABLE_SEACASNemspread:BOOL=ON' '-DTrilinos_ENABLE_SEACASNemslice:BOOL=ON' '-DTrilinos_ENABLE_SEACASChaco:BOOL=OFF' '-DTrilinos_ENABLE_SEACASNemslice:BOOL=OFF' '-DTrilinos_ENABLE_Thyra:BOOL=ON' '-DTrilinos_ENABLE_ThyraEpetraAdapters:BOOL=ON' '-DTrilinos_ENABLE_ThyraEpetraExtAdapters:BOOL=ON' '-DTrilinos_ENABLE_ThyraTpetraAdapters:BOOL=ON' '-DTPL_ENABLE_BLAS:BOOL=ON' '-DBLAS_LIBRARY_NAMES:STRING=openblas' '-DBLAS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/openblas-0.3.15-3uzmsi2jrh4o7rcrys4wa2tg6ovki55t/lib' '-DTPL_ENABLE_LAPACK:BOOL=ON' '-DLAPACK_LIBRARY_NAMES:STRING=openblas' '-DLAPACK_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/openblas-0.3.15-3uzmsi2jrh4o7rcrys4wa2tg6ovki55t/lib' '-DTPL_ENABLE_GLM:BOOL=ON' '-DTPL_ENABLE_Matio:BOOL=ON' '-DTPL_ENABLE_X11:BOOL=OFF' '-DTrilinos_ENABLE_Gtest:BOOL=ON' '-DTPL_ENABLE_Netcdf:BOOL=ON' '-DNetCDF_ROOT:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/netcdf-c-4.8.0-rudvlpfj65jss3v6byafnuqzctjx2e7f' '-DTPL_ENABLE_HYPRE:BOOL=ON' '-DHYPRE_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hypre-2.20.0-btbzpjcwalrwtdk62idk2qzzjioeylxs/include' '-DHYPRE_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hypre-2.20.0-btbzpjcwalrwtdk62idk2qzzjioeylxs/lib' '-DTPL_ENABLE_Boost:BOOL=ON' '-DBoost_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/boost-1.76.0-xfahzpfzjei2i62ni7seuszkjwfzmvjz/include' '-DBoost_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/boost-1.76.0-xfahzpfzjei2i62ni7seuszkjwfzmvjz/lib' '-DTPL_ENABLE_HDF5:BOOL=ON' '-DHDF5_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hdf5-1.12.0-zlpe7nx7g3rt4sthx36aoqzysrwyrjg4/include' '-DHDF5_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hdf5-1.12.0-zlpe7nx7g3rt4sthx36aoqzysrwyrjg4/lib' '-DTPL_ENABLE_Cholmod:BOOL=OFF' '-DTPL_ENABLE_UMFPACK:BOOL=ON' '-DUMFPACK_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/suite-sparse-5.10.1-375laj47cael7c3xlphppxz6ljxos3vp/lib' '-DUMFPACK_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/suite-sparse-5.10.1-375laj47cael7c3xlphppxz6ljxos3vp/include' '-DUMFPACK_LIBRARY_NAMES:STRING=umfpack;amd;colamd;cholmod;suitesparseconfig' '-DTPL_ENABLE_METIS:BOOL=ON' '-DTPL_ENABLE_ParMETIS:BOOL=ON' '-DMETIS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/metis-5.1.0-lxkd2thzi5xc5z3aqvw4eodot57r53y5/lib' '-DMETIS_LIBRARY_NAMES:STRING=metis' '-DTPL_METIS_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/metis-5.1.0-lxkd2thzi5xc5z3aqvw4eodot57r53y5/include' '-DTPL_ENABLE_ParMETIS:BOOL=ON' '-DParMETIS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/parmetis-4.0.3-rdrwwqrizoiz4h3xiehqqwfzimpuv5zh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/metis-5.1.0-lxkd2thzi5xc5z3aqvw4eodot57r53y5/lib' '-DParMETIS_LIBRARY_NAMES:STRING=parmetis;metis' '-DTPL_ParMETIS_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/parmetis-4.0.3-rdrwwqrizoiz4h3xiehqqwfzimpuv5zh/include;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/metis-5.1.0-lxkd2thzi5xc5z3aqvw4eodot57r53y5/include' '-DTPL_ENABLE_MUMPS:BOOL=ON' '-DTPL_ENABLE_SCALAPACK:BOOL=ON' '-DMUMPS_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mumps-5.4.0-vy5bpjrdcgs63m47fjy3fbfxgo6frnel/lib' '-DMUMPS_LIBRARY_NAMES:STRING=dmumps;mumps_common;pord' '-DSCALAPACK_LIBRARY_NAMES:STRING=scalapack' '-DSCALAPACK_LIBRARY_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/netlib-scalapack-2.1.0-ryeq4ydyoueo2triza7dhwg6ihkkks46/lib' '-DTPL_ENABLE_SuperLUDist:BOOL=OFF' '-DTPL_ENABLE_SuperLU:BOOL=OFF' '-DTPL_ENABLE_STRUMPACK:BOOL=OFF' '-DTPL_ENABLE_Pnetcdf:BOOL=OFF' '-DTPL_ENABLE_Zlib:BOOL=OFF' '-DTPL_ENABLE_CGNS:BOOL=OFF' '-DTPL_ENABLE_ADIOS2:BOOL=OFF' '-DKokkos_ARCH_:BOOL=ON' '-DTPL_ENABLE_CUDA:BOOL=OFF' '-DCMAKE_CXX_STANDARD:STRING=11' '-DTrilinos_ENABLE_OpenMP:BOOL=OFF' '-DTrilinos_ENABLE_Fortran:BOOL=ON' '-DTrilinos_EXTRA_LINK_FLAGS:STRING=-L/usr/lib/gcc/x86_64-linux-gnu/9/ -lgfortran' '-DTeuchos_ENABLE_COMPLEX:BOOL=OFF' '-DTeuchos_ENABLE_FLOAT:BOOL=OFF' '-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON' '-DTpetra_INST_DOUBLE:BOOL=ON' '-DTpetra_INST_COMPLEX_DOUBLE:BOOL=OFF' '-DTpetra_INST_COMPLEX_FLOAT:BOOL=OFF' '-DTpetra_INST_FLOAT:BOOL=OFF' '-DTpetra_INST_SERIAL:BOOL=ON' '-DTpetra_INST_INT_INT:BOOL=OFF' '-DTpetra_INST_INT_LONG:BOOL=ON' '-DTpetra_INST_INT_LONG_LONG:BOOL=OFF' '-DCMAKE_INSTALL_NAME_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj/lib' '-DCMAKE_CXX_FLAGS:STRING=-DMUMPS_5_0' '-DTrilinos_ENABLE_Pike:BOOL=OFF' '/tmp/root/spack-stage/spack-stage-trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj/spack-src'

1 error found in build log:
     914    -- SEACASSupes_exttest: NOT added test because SEACASSupes_ENABLE_TESTS=', SEACAS_ENABLE_TESTS='''.
     915    -- A Python-2 version of exodus.py and exomerge.py will be installed.
     916    Processing enabled package: Anasazi (Libs)
     917    Processing enabled package: Ifpack2 (Libs)
     918    Processing enabled package: Stratimikos (Libs)
     919    Processing enabled package: Teko (Libs)
  >> 920    CMake Error at packages/teko/CMakeLists.txt:25 (MESSAGE):
     921      Teko currently requires that Tpetra enable either GlobalOrdinal (GO) = long
     922      long or GO = int.  Please set Tpetra_INST_INT_LONG_LONG=ON to enable GO =
     923      long long, or Tpetra_INST_INT_INT=ON to enable GO = int.  If you would like
     924      to use other GO types, please contact the Teko developers.
     925
     926

See build log for details:
  /tmp/root/spack-stage/spack-stage-trilinos-13.0.1-wtf4tfw37ckqd2lvh27umelqfg37asdj/spack-build-out.txt
```

It seems like the resolution is pretty straightforward. Recording here to keep track as it is resolved.

FYI @sethrj @keitat

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -100,7 +100,7 @@ class Trilinos(CMakePackage, CudaPackage):
             description='Enable ADIOS2')
     variant('glm',          default=True,
             description='Compile with GLM')
-    variant('gtest',        default=True,
+    variant('gtest',        default=False,
             description='Compile with Gtest')
     variant('hdf5',         default=True,
             description='Compile with HDF5')

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -306,6 +306,7 @@ class Trilinos(CMakePackage, CudaPackage):
     conflicts('+teko', when='~ml')
     conflicts('+teko', when='~teuchos')
     conflicts('+teko', when='~tpetra')
+    conflicts('+teko', when='gotype=long')
     conflicts('+tempus', when='~nox')
     conflicts('+tempus', when='~teuchos')
     conflicts('+tpetra', when='~kokkos')

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -5,6 +5,7 @@
 
 import os
 import sys
+
 from spack import *
 from spack.operating_systems.mac_os import macos_version
 from spack.pkg.builtin.kokkos import Kokkos

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -29,7 +29,7 @@ class Trilinos(CMakePackage, CudaPackage):
     url      = "https://github.com/trilinos/Trilinos/archive/trilinos-release-12-12-1.tar.gz"
     git      = "https://github.com/trilinos/Trilinos.git"
 
-    maintainers = ['keitat']
+    maintainers = ['keitat', 'sethrj']
 
     # ###################### Versions ##########################
 

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -70,7 +70,7 @@ class Trilinos(CMakePackage, CudaPackage):
     variant('float', default=False,
             description='Enable single precision (float) numbers in Trilinos')
     variant('gotype', default='long',
-            values=('none', 'int', 'long', 'long_long'),
+            values=('int', 'long', 'long_long', 'all'),
             multi=False,
             description='global ordinal type for Tpetra')
     variant('fortran',      default=True,
@@ -370,12 +370,15 @@ class Trilinos(CMakePackage, CudaPackage):
     conflicts('+scorec', when='~stk')
     conflicts('+scorec', when='~zoltan')
 
+    # Multi-value gotype only applies to trilinos through 12.14
+    conflicts('gotype=all', when='@12.15:')
+
     # All compilers except for pgi are in conflict:
     for __compiler in spack.compilers.supported_compilers():
         if __compiler != 'clang':
             conflicts('+cuda', when='~wrapper %{0}'.format(__compiler),
-                      msg='trilinos~wrapper+cuda can only be built with the\
-                      Clang compiler')
+                      msg='trilinos~wrapper+cuda can only be built with the '
+                      'Clang compiler')
 
     # ###################### Dependencies ##########################
 
@@ -876,21 +879,13 @@ class Trilinos(CMakePackage, CudaPackage):
             ])
 
             gotype = spec.variants['gotype'].value
-            # default in older Trilinos versions to enable multiple GOs
-            if ((gotype == 'none') and spec.satisfies('@:12.14.1')):
+            if gotype == 'all':
+                # default in older Trilinos versions to enable multiple GOs
                 options.extend([
-                    '-DTpetra_INST_INT_INT:BOOL=ON',
-                    '-DTpetra_INST_INT_LONG:BOOL=ON',
-                    '-DTpetra_INST_INT_LONG_LONG:BOOL=ON'
+                    define('Tpetra_INST_INT_INT', True),
+                    define('Tpetra_INST_INT_LONG', True),
+                    define('Tpetra_INST_INT_LONG_LONG', True),
                 ])
-            # set default GO in newer versions to long
-            elif (gotype == 'none'):
-                options.extend([
-                    '-DTpetra_INST_INT_INT:BOOL=OFF',
-                    '-DTpetra_INST_INT_LONG:BOOL=ON',
-                    '-DTpetra_INST_INT_LONG_LONG:BOOL=OFF'
-                ])
-            # if another GO is specified, use this
             else:
                 options.extend([
                     define('Tpetra_INST_INT_INT', gotype == 'int'),

